wordpress-mobile/WordPress-Android
# Make GCM message service not static

Fixes #10626

GCMMessageService had static array of notifications and a static Notification Handler. This approach made it impossible to use any injected field (and is generally not great). I'm taking a simple step in this PR where I'm creating a GCMMessageHandler which is an injected Singleton. This class contains all the static code from GCMMessageService (without changes) and is injected wherever the GCMMessageService was used. This felt like a good and simple first step. 

I've added a notification shown tracking to the GCMMessageHandler.

To test:
* Check that the notifications work as they used to 
* Have a notification
* Check that LogCat shows a `notification_shown` event with a `notification_type` 

PR submission checklist:
